integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/PerformanceResource.java

```diff
@@ -1,50 +1,43 @@
 package com.redhat.service.smartevents.integration.tests.resources;
 
-import com.redhat.service.smartevents.integration.tests.common.Constants;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
-import io.restassured.response.Response;
+import software.tnb.hyperfoil.validation.HyperfoilValidation;
+import software.tnb.hyperfoil.validation.generated.model.Run;
 
 public class PerformanceResource {
 
     private static final String HYPERFOIL_URL = System.getProperty("performance.hyperfoil.url");
-    private static final String BASE_BENCHMARK_URL = HYPERFOIL_URL + "/benchmark/";
-    private static final String BASE_RUN_URL = HYPERFOIL_URL + "/run/";
-
-    public static void addBenchmark(String managerToken, String benchmarkContent, String contentType) {
-        createBenchmarkResponse(managerToken, benchmarkContent, contentType)
-                .then()
-                .log().ifValidationFails()
-                .statusCode(204);
+    private static final String BENCHMARK_CONTENT_FILE = "benchmark.hf.yaml";
+
+    // Creating validation object directly as TNB Hyperfoil default service starts/stops Hyperfoil, causing tests to fail on environment without Docker environment
+    public static HyperfoilValidation validation = new HyperfoilValidation(HYPERFOIL_URL);
+
+    public static void addBenchmark(String benchmarkContent, String contentType) {
+        // Currently TNB expects the benchmark to be provided either on classpath or as URL, so storing the benchmark content in dedicated file
+        try {
+            URL benchmarkResource = PerformanceResource.class.getClassLoader().getResource(BENCHMARK_CONTENT_FILE);
+            Path benchmarkResourcePath = Paths.get(benchmarkResource.toURI());
+            Files.writeString(benchmarkResourcePath, benchmarkContent);
+        } catch (URISyntaxException | IOException e) {
+            throw new RuntimeException("Exception while storing benchmark content into file on classpath", e);
+        }
+
+        validation.addBenchmark(BENCHMARK_CONTENT_FILE, null);
```

Comment:
if we're not currently passing any value as host target value, why not just use `validation.addBenchmark(BENCHMARK_CONTENT_FILE)` instead?

Reply:
Good point, will change it

Reply:
Done